genume/registry/xparser.py

```diff
@@ -13,26 +14,42 @@ def parse(cmdl, c):
             # basic value case.
             e = cmdl.pop(0)
             k = cmdl.pop(0)
-            v = ValueEntry(c, cmdl.pop(0))
+            v = ValueEntry(c, cmdl.pop(0).strip())
             if e == "ADV":
                 v.level = InfoLevel.advanced
             c[k] = v
+        elif cmd == "VALUES":
+            # entry of list
+            e = cmdl.pop(0)
+            k = cmdl.pop(0)
+            v = cmdl.pop(0).strip()
+
+            if k in c:
+                c[k].add(v)
+            else:
+                e = ListEntry(c, v)
+                if e == "ADV":
+                    e.level = InfoLevel.advanced
+                c[k] = e
+
         else:
             print("Warn: invalid command {0}".format(cmd))
 
+
 def run(f):
     if os.access(str(f), os.X_OK):
         o = sbpr.check_output(str(f))
-        # Make everything be on a single line.
-        s = o.decode("utf-8").strip().replace("\n", " ")
```

Comment:
@Sima214 Did you removed the new lines for a reason?

Reply:
@MaanooAk I was removing the new lines, because I am passing the output to a function which splits a shell command into arguments.
If it handles new lines in the expected way then it is okay.
Anyways xparser will eventually need to be implemented with regex, so this won't be a problem. Do we have anyone who is really good at regex? I am okay at it.

PS: Please rename processors to threads. It triggers me every time I read it.

Reply:
Seems to be working. Regex will not improve the performance and shlex seems to be enough. Do you have something in mind that can be parsed without regex?

PS: I come from windows, where threads are actually children of processes.

Reply:
Checked shlex's implementation and it is pretty much what we want. Also waiting for new processes to spawn will be by far the slowest part of this (not tested).